vespa/package.py

```diff
@@ -889,6 +890,156 @@ def __init__(
+                    "input_ids": "input_ids",
+                    "token_type_ids": "token_type_ids",
+                    "attention_mask": "attention_mask",
+                },
+                outputs={"output_0": "logits"},
+            )
+        )
+
+        #
+        # Add query profile type for query token ids
+        #
+        self.query_profile_type.add_fields(
+            QueryTypeField(
+                name="ranking.features.query({})".format(
+                    model_config.query_token_ids_name
+                ),
+                type="tensor<float>(d0[{}])".format(
+                    int(model_config.actual_query_input_size)
+                ),
+            )
+        )
+
+        #
+        # Add field for doc token ids
+        #
+        if not doc_token_ids_indexing:
+            doc_token_ids_indexing = ["attribute", "summary"]
+        self.schema.add_fields(
+            Field(
+                name=model_config.doc_token_ids_name,
+                type="tensor<float>(d0[{}])".format(
+                    int(model_config.actual_doc_input_size)
+                ),
+                indexing=doc_token_ids_indexing,
+            ),
+        )
+
+        #
+        # Add rank profiles
+        #
+        constants = {"TOKEN_NONE": 0, "TOKEN_CLS": 101, "TOKEN_SEP": 102}
+        if "contants" in kwargs:
+            constants.update(kwargs.pop("contants"))
+
+        functions = [
+            Function(
+                name="question_length",
+                expression="sum(map(query({}), f(a)(a > 0)))".format(
+                    model_config.query_token_ids_name
+                ),
+            ),
+            Function(
+                name="doc_length",
+                expression="sum(map(attribute({}), f(a)(a > 0)))".format(
+                    model_config.doc_token_ids_name
+                ),
+            ),
+            Function(
+                name="input_ids",
+                expression="tokenInputIds({}, query({}), attribute({}))".format(
+                    model_config.input_size,
+                    model_config.query_token_ids_name,
+                    model_config.doc_token_ids_name,
+                ),
+            ),
+            Function(
+                name="attention_mask",
+                expression="tokenAttentionMask({}, query({}), attribute({}))".format(
+                    model_config.input_size,
+                    model_config.query_token_ids_name,
+                    model_config.doc_token_ids_name,
+                ),
+            ),
+            Function(
+                name="token_type_ids",
+                expression="tensor<float>(d0[1],d1[{}])(\n"
+                "    if (d1 < question_length + 2,\n"
+                "        0,\n"
+                "    if (d1 < question_length + doc_length + 3,\n"
+                "        1,\n"
+                "        TOKEN_NONE\n"
+                "    )))".format(model_config.input_size),
+            ),
```

Comment:
I will update `token_type_uds` to use `tokenTypeIds` as soon as https://github.com/vespa-engine/vespa/pull/16413 is available.

Reply:
Now, https://github.com/vespa-engine/vespa/pull/16432

Reply:
I will change that in a separate PR upon Vespa 7.355.9 release.